examples/extensible-interface-files/core-plugins/basic-example/README.md

```diff
@@ -0,0 +1,94 @@
+module.
+
+This example project intends to show the process of writing the core interface field, loading it
+within the `IfG a`/`IfL a` interface global/local loading environments, and finally using that data
+to lookup the right-hand sides of external names that appear in the core that is given in the plugin
+environment of the currently compiling module.
+
+## Deliverables
+
+* Example structure of a plugin using extensible interface files and the `-fwrite-core-field` flag
+* Usage of loading the `IfG`/`IfL` (interface load) monad environments into `IO`, which can then be
+  lifted into the plugin action via `MonadIO`
+* Inlining the plugin binds based on the core recovered from interface files of dependencies
+
+## Design and Implementation
+
+We define a `Lib` module, which contains stand-in definitions representing external bindings that
+would require being looked up if their names appeared in the right-hand sides of the definitions
+in another module. This module is compiled with the `-fwrite-core-field` flag, so its interface
+file will contain a field that can be deserialised into a `ModGuts`.
+
+Then, the `Lib` module is imported into the `Main` module, which then makes use of these imported
+definitions in its own definitions. `Main` is then compiled with the `Example` plugin, which
+performs a version of inlining to immitate Plutus. Because `Lib`'s definitions are external to
+`Main`, when the plugin encounters a name from `Lib`, it will have to recover the core from `Lib.hs`'s
+core extensible interface field.
+
+The `Example` module, containing the post-type checking plugin, makes use of the core found in
+interface files to perform a form of inlining to demonstrate usage for Plutus. The current module's
+bindings are found in the plugin argument, `TcGblEnv`, and the `TcM` monad in which the plugin
+operates gives us access to other environment data that we require to load interfaces.
+
+## Details
+
+* Loading interface files is done in the `type IfL a = IOEnv (Env IfGbl IfLcl) a` monad, while
+  core plugins are in the `CoreM` enviroment. `IfL` is meant to be eventually run in `IO`, and
+  `CoreM` is `MonadIO`, so it can be lifted into the plugin, but initialising the interface loading
+  does require some extra data:
+  * We must first initialise the global enviroment, which requires the `HscEnv`, which can be
+    retrieved from `getHscEnv`. This `HscEnv` differs based on the current module being compiled
+  * For each external module that is being imported, the local enviroment must be initialised with
+    its `Module` data structure, essentially setting that one as active. These `Module`s are able
+    to be recovered from the module graph contained in the `HscEnv`
+
+## Building
+
+### GHC
+
+With https://gitlab.haskell.org/ghc/ghc/-/tree/wip/coreField run:
+
+./boot
+./configure
+make -j8
+sudo make install
+
+Then include /usr/local/bin in the path.

```

Comment:
Alternatively (and so that you don't mess with your global install) you can also do:
```
/path/to/ghc/inplace/bin/ghc-stage2 Main.hs ...
```

So either
```
sudo make install
export PATH="/usr/local/bin:$PATH"
export HC=$(which ghc)
```
or
```
export HC=/path/to/ghc/inplace/bin/ghc-stage2
```

this migh mess with ghcid though :-/

You can also control the path where ghc will end up being installed by providing `--prefix=/my/ghc/intall/path` to `configure`.

Installing custom compilers globally easily leads to conflicts.

Reply:
I'll make this more clear. I've been temporarily putting it into the path, but I didn't include it because fish shell has different syntax.